includes/islandora_scan_batch.inc

```diff
@@ -188,9 +188,20 @@ class IslandoraScanBatch extends IslandoraBatchPreprocessor {
       $object->addRelationships();
       $this->addToDatabase($object, $object->getResources(), $parent);
       $to_return[] = $object;
-
-      foreach ($object->getChildren($this->connection) as $child) {
-        $to_return = array_merge($to_return, $this->preprocessChildren($child, $object->id));
+      $children = $object->getChildren($this->connection);
+      $child_count = count($children);
+      if ($child_count > 0) {
+        if (isset($this->parameters['namespace'])) {
+          $namespace = $this->parameters['namespace'];
+        }
+        else {
+          $namespace = $this->determineNamespace($object);
+        }
+        $children_ids = $this->connection->repository->getNextIdentifier($namespace, NULL, $child_count);
+        foreach ($children_ids as $key => $id) {
```

Comment:
Maybe i'm a bit overconcern with this, so it's just a comment: Imports are a sensitive topic and when things get wrong they get wrong Big. Would it not be better to iterate over $children instead of $children_ids? children_ids is hitting directly the repository and I can see edge cases (unprovable but still hard to debug) around that could make ` $this->connection->repository->getNextIdentifier()` respond weirdly which would make objects (children) being silently dropped. If the iteration is over the children itself, then on missing PID's for a certain child, there could be a fallback or even an exception. Does that makes any sense?